match/post_ohr.py

```diff
@@ -0,0 +1,333 @@
+
+    dfb = pprr[["uid", "first_name", "last_name", "agency", "middle_name"]]
+    dfb = dfb.drop_duplicates(subset=["uid"]).set_index("uid")
+    dfb.loc[:, "fc"] = dfb.first_name.fillna("").map(lambda x: x[:1])
+    dfb.loc[:, "lc"] = dfb.last_name.fillna("").map(lambda x: x[:1])
+
+    matcher = ThresholdMatcher(
+        ColumnsIndex(["fc"]),
+        {
+            "first_name": JaroWinklerSimilarity(),
+            "last_name": JaroWinklerSimilarity(),
+            "agency": JaroWinklerSimilarity(),
+        },
+        dfa,
+        dfb,
+    )
+    decision = 0.950
+
+    matches = matcher.get_index_pairs_within_thresholds(decision)
+    match_dict = dict(matches)
+
+    # value is added to "extracted_uid" column if match score is 1
+    cprr.loc[:, "matched_uid_7"] = cprr.uid_7.map(lambda x: match_dict.get(x))
+    return cprr
+
+
+def match_uid_8_and_personnel(cprr, pprr):
+    dfa = cprr[["uid_8", "first_name", "last_name", "agency_8", "middle_name"]]
+    dfa = dfa.rename(columns={"agency_8": "agency"})
+    dfa = dfa.drop_duplicates(subset=["uid_8"]).set_index("uid_8")
+    dfa.loc[:, "fc"] = dfa.first_name.fillna("").map(lambda x: x[:1])
+    dfa.loc[:, "lc"] = dfa.last_name.fillna("").map(lambda x: x[:1])
+
+    dfb = pprr[["uid", "first_name", "last_name", "agency", "middle_name"]]
+    dfb = dfb.drop_duplicates(subset=["uid"]).set_index("uid")
+    dfb.loc[:, "fc"] = dfb.first_name.fillna("").map(lambda x: x[:1])
+    dfb.loc[:, "lc"] = dfb.last_name.fillna("").map(lambda x: x[:1])
+
+    matcher = ThresholdMatcher(
+        ColumnsIndex(["fc"]),
+        {
+            "first_name": JaroWinklerSimilarity(),
+            "last_name": JaroWinklerSimilarity(),
+            "agency": JaroWinklerSimilarity(),
+        },
+        dfa,
+        dfb,
+    )
+    decision = 0.9
+
+    matches = matcher.get_index_pairs_within_thresholds(decision)
+    match_dict = dict(matches)
+
+    # value is added to "extracted_uid" column if match score is 1
+    cprr.loc[:, "matched_uid_8"] = cprr.uid_8.map(lambda x: match_dict.get(x))
+    return cprr
+
+
+def join_matched_uids(df):
+    df.loc[:, "post_id"] = df.matched_uid.fillna("").str.cat(
+        df.matched_uid_1.fillna(""), sep=","
+    )
+    df.loc[:, "post_id"] = df.post_id.str.cat(df.matched_uid_2.fillna(""), sep=",")
+    df.loc[:, "post_id"] = df.post_id.str.cat(df.matched_uid_3.fillna(""), sep=",")
+    df.loc[:, "post_id"] = df.post_id.str.cat(df.matched_uid_4.fillna(""), sep=",")
+    df.loc[:, "post_id"] = df.post_id.str.cat(df.matched_uid_5.fillna(""), sep=",")
+    df.loc[:, "post_id"] = df.post_id.str.cat(df.matched_uid_6.fillna(""), sep=",")
+    df.loc[:, "post_id"] = df.post_id.str.cat(df.matched_uid_7.fillna(""), sep=",")
+    df.loc[:, "post_id"] = df.post_id.str.cat(df.matched_uid_8.fillna(""), sep=",")
+
+    df.loc[:, "post_id"] = (
+        df.post_id.fillna("")
+        .str.replace(r"\,\, ?(.+)?", "", regex=True)
+        .str.replace(r"^\,", "", regex=True)
+    )
```

Comment:
You're overusing regex. Regex can be very slow so use them sparingly. For the purpose of joining columns, just use `DataFrame.apply`:
```python
df.loc[:, 'post_id'] = df[
    [col for col in df.columns if col.startswith('matched_uid_')]
].apply(lambda sr: ','.join(sr[sr.notna()].to_list()))
```
```

Reply:
Thanks